### ISSUE TITLE
Core Name doesn't support Japanese characters.

### ISSUE THREAD

**Issue by DarthSidious666** [m64p/m64p#297]
![image](https://user-images.githubusercontent.com/57962477/187702400-34e884ed-96e5-452d-8791-730398de1264.png)

As shown in the image the Japanese characters are shown as boxes.
It should be showing this: 64で発見!!たまごっち みんなでたまごっちワールド

Roughly translated as Tamagotchi World 64.

**Comment by loganmc10** [comment]
Yeah it's kind of a hard problem to solve, the GoodName should show correctly, for instance:
```
Core: Using full mem base
Core: Goodname: AI Shougi 3 (J) [!]
Core: Name: AI�����3            
```
The problem is that the Core name (the name inside the ROM header) is encoded using Shift-JIS, not Utf8. I don't know of a way to decode Shift-JIS text in Qt6. Qt5 has a library that can do it (QTextCodec), but it was dropped in Qt6.

The same problem occurs in mupen64plus-ui-console.

**Comment by loganmc10** [comment]
This is the bug in Qt for reference: https://bugreports.qt.io/browse/QTBUG-86437

It seems that their viewpoint is that everybody uses UTF these days (including for Japanese), which is true, but doesn't help when you're trying to display text created in the 1990's :)

**Comment by DarthSidious666** [comment]
Would it be easier to remove the core name then since most romsets use the goodname for verification.
I think you would have to parse the string to convert to UTF.

According to this: https://bugreports.qt.io/browse/QTBUG-103375 it might be making a return in qt6.4, but I'm not sure.

**Comment by loganmc10** [comment]
That output is coming from mupen64plus-core. The goodname comes from mupen64plus.ini, so it's probably still good to show the name reported by the ROM header when it's possible (for english ROMs)

**Comment by DarthSidious666** [comment]
Think I've found a solution from here for the string parsing: https://stackoverflow.com/questions/65379825/qtcore-qtextcodec-not-found-in-qt-6

**Comment by loganmc10** [comment]
I'd rather not use the Qt5 compatibility library, it would pull in some extra .dll's, just to fix this problem.

They say QStringConverter is the replacement library in Qt6, but QStringConverter doesn't support encodings besides UTF, so it won't be able to decode Shift JIS text (at least in 6.3).

**Comment by DarthSidious666** [comment]
Rats.

I'll keep trying to find a solution as it would be nice for it to be supported even if it's a minor issue.
